docs/developer/decision-records/2022-06-03-event-framework/README.md

```diff
@@ -0,0 +1,98 @@
+# Event framework
+
+## Decision
+
+This decision record covers concepts and semantics about asynchronous communication in the EDC.
+This is needed to define what's an `Event` and how this concept should be introduced into EDC.
+
+## Rationale
+
+There are two asynchronous communication levels in EDC:
+
+#### Protocol level
+Every call from a connector to another regards the communication protocol, currently the only protocol implemented in EDC is IDS.
+A protocol call could have an asynchronous response, and that's called **callback**.
+
+#### Instance level
+The EDC should have the possibility to communicate with third party systems (known as subscribers) notifying them about what's happening in the connector. This communication needs to be made through **events**.
+An event is a representation of something that happened in the past, and has very specific characteristics, like:
+- it's named with a short sentence in the past tense (e.g. "contract negotiation confirmed")
+- it's serializable
+- it contains the issue timestamp
+- it describes which layer emitted the event: domain, system, security, and so on...
+
+#### Main differences between Events and Callbacks
+- Callbacks are always correlated with a request, events don't
+- Callbacks are unicast, events are multicast
+- Callbacks are at the protocol level, events can be at a different level
+
```

Comment:
I would also add that:
"Callbacks are critical for the correct operation of a connector, where events are optional"